components/properties/src/script.rs

```diff
@@ -109,14 +128,20 @@ impl<'data> ScriptExtensions<'data> {
         } else if sc_with_ext.is_inherited() {
             Script::Inherited
         } else {
-            let script_val = sc_with_ext.0 & SCRIPT_X_SCRIPT_VAL;
+            let script_val = sc_with_ext.0;
             Script(script_val)
         }
     }
 
-    pub fn get_script_extensions_val(&self, code_point: u32) -> &ZeroSlice<Script> {
-        let sc_with_ext = self.trie.get(code_point);
-
+    // Returns the Script_Extensions value for a code_point when the trie value
+    // is already known.
+    // This private helper method exists to prevent code duplication in callers like
+    // `get_script_extensions_val`, `get_script_extensions_set`, and `has_script`.
+    fn get_scx_val_using_trie_val(
+        &self,
+        code_point: u32,
```

Comment:
Yes.  The lookup in the last branch is done because we need the `ScriptWithExt` by reference rather than by value, so that we can cast it to a slice-of-length-one.  However, we can solve this problem by making this function have a signature like

```rust
fn get_scx_val_using_trie_val(
        &self,
        sc_with_ext: &ScriptWithExt::ULE,
)
```

The call sites would need to change to pass the ScriptWithExt by reference rather than by value.

This will also be a little less convenient because you need to manually invoke `ScriptWithExt::from_unaligned()`.  But that's a cost I'm willing to pay in order to avoid the double lookup.